frontend/src/layout/Header.tsx

```diff
@@ -62,85 +64,255 @@ const Search: React.FC = () => {
+                <Link to="/about" css={itemStyle} onClick={closeMenu}>
+                    <FontAwesomeIcon icon={faInfoCircle} fixedWidth css={iconStyle} />
+                    {t("main-menu.about")}
+                </Link>
+            </li>
+        </ul>
+    </>;
+};
+
+const LoginButton: React.FC<{ closeMenu: () => void }> = ({ closeMenu }) => {
+    const { t } = useTranslation();
+
+    return (
+        <Link
+            to="/login"
+            onClick={closeMenu}
+            css={{
+                display: "inline-block",
+                marginTop: 8,
+                padding: "4px 8px",
+                fontSize: 18,
+                textDecoration: "none",
+                whiteSpace: "nowrap",
+                border: "2px solid #495dce",
+                borderRadius: 3,
+                "&:hover": {
+                    boxShadow: "0 0 5px #0000004d",
+                },
+            }}
+        >
+            {t("login.button-label")}
+        </Link>
+    );
+};
+
 const LanguageList = () => {
     const { t, i18n } = useTranslation();
 
     return (
         <ul css={{
             width: "100%",
             listStyle: "none",
+            fontSize: 18,
             margin: 0,
             padding: 0,
         }}>
-            {Object.keys(languages).map(lng => (
-                <li
-                    onClick={() => i18n.changeLanguage(lng)}
-                    key={lng}
-                    css={{
-                        padding: "4px 8px 4px 12px",
-                        cursor: "pointer",
-                        "&:hover": {
-                            backgroundColor: "#ddd",
-                        },
-                    }}
-                >
-                    <div className="fa-fw" css={{ display: "inline-block", marginRight: 8 }}>
-                        {lng === i18n.language && <FontAwesomeIcon icon={faCheck} fixedWidth />}
-                    </div>
-                    {t("language-name", { lng })}
-                </li>
-            ))}
+            {Object.keys(languages).map(lng => {
+                // We check whether a language is active by simply translating
+                // with the automatically determined language and with the
+                // specific one. If the result is the same, that's the current
+                // language.
+                //
+                // I do realize that it could be that one language "foo" has
+                // only the string "language-name" translated and thus, the
+                // whole page would be shown with the fallback language (en),
+                // but the language menu would still show "foo" as active
+                // language. Or imagine the other way around: everything except
+                // "language-name" is translated. But both of these cases should
+                // be avoided anyway when creating a translation.
+                const isActive = t("language-name", { lng }) === t("language-name");
```

Comment:
You know about [`i18next.language`](https://www.i18next.com/overview/api#language) and [`i18next.languages`](https://www.i18next.com/overview/api#languages), right?

If so, I would expect the comment to say something about why you don't use those APIs.